test/python/circuit/test_circuit_operations.py

```diff
@@ -440,6 +440,22 @@ def test_measure_all(self):
 
         self.assertEqual(expected, circuit)
 
+    def test_measure_all_not_add_creg(self):
+        """Test measure_all applies measurements to all qubits.
+        Does not create a new ClassicalRegister if the existing one is big enough.
+        """
+        qr = QuantumRegister(2)
+        cr = ClassicalRegister(2, "meas")
+
+        circuit = QuantumCircuit(qr, cr)
+        circuit.measure_all(add_creg=False)
+
+        expected = QuantumCircuit(qr, cr)
+        expected.barrier()
+        expected.measure(qr, cr)
+
+        self.assertEqual(expected, circuit)
+
```

Comment:
This is a good test - it's very clear.

Please could you also add a test that it works if there are more classical bits than qubits, and a test that the correct error is raised if you try to use `add_bits=False` without enough classical bits?